Remove :role() pseudo-class implementation

nt1m opened the pull request:
#### 4f6fdd5fa416190cbc901af7dc4407c40ed9c03e
<pre>
Remove :role() pseudo-class implementation
<a href="https://bugs.webkit.org/show_bug.cgi?id=243151">https://bugs.webkit.org/show_bug.cgi?id=243151</a>

Reviewed by Ryosuke Niwa.

The matching is not implemented, and it&apos;s not part of the spec: <a href="https://github.com/w3c/csswg-drafts/issues/3596.">https://github.com/w3c/csswg-drafts/issues/3596.</a>
Let&apos;s remove since it&apos;s confusing.

* Source/WebCore/css/CSSSelector.cpp:
(WebCore::appendPseudoClassFunctionTail):
(WebCore::CSSSelector::selectorText const):
* Source/WebCore/css/CSSSelector.h:
* Source/WebCore/css/SelectorChecker.cpp:
(WebCore::SelectorChecker::checkOne const):
* Source/WebCore/css/SelectorPseudoClassAndCompatibilityElementMap.in:
* Source/WebCore/css/parser/CSSSelectorParser.cpp:
(WebCore::isOnlyPseudoClassFunction):
(WebCore::CSSSelectorParser::consumePseudo):
* Source/WebCore/cssjit/SelectorCompiler.cpp:
(WebCore::SelectorCompiler::addPseudoClassType):

* LayoutTests/fast/css/css-selector-text-expected.txt:
* LayoutTests/fast/css/css-selector-text.html:
* LayoutTests/fast/css/css-set-selector-text-expected.txt:
* LayoutTests/fast/css/css-set-selector-text.html:
* LayoutTests/fast/selectors/invalid-functional-pseudo-class-expected.txt:
* LayoutTests/fast/selectors/invalid-functional-pseudo-class.html:

Canonical link: <a href="https://commits.webkit.org/252786@main">https://commits.webkit.org/252786@main</a>
</pre>

rniwa commented on LayoutTests/fast/css/css-selector-text.html:
```diff
@@ -703,24 +703,6 @@
-shouldThrow("parseThenSerializeRule(':role(:role(a)) { }')");
-
-debug('');
-
```
We should probably add a test case for `:role(~)` to fail parsing.

rniwa reviewed (approved):
r=me

nt1m committed:
Remove :role() pseudo-class implementation
https://bugs.webkit.org/show_bug.cgi?id=243151

Reviewed by Ryosuke Niwa.

The matching is not implemented, and it's not part of the spec: https://github.com/w3c/csswg-drafts/issues/3596.
Let's remove since it's confusing.

* Source/WebCore/css/CSSSelector.cpp:
(WebCore::appendPseudoClassFunctionTail):
(WebCore::CSSSelector::selectorText const):
* Source/WebCore/css/CSSSelector.h:
* Source/WebCore/css/SelectorChecker.cpp:
(WebCore::SelectorChecker::checkOne const):
* Source/WebCore/css/SelectorPseudoClassAndCompatibilityElementMap.in:
* Source/WebCore/css/parser/CSSSelectorParser.cpp:
(WebCore::isOnlyPseudoClassFunction):
(WebCore::CSSSelectorParser::consumePseudo):
* Source/WebCore/cssjit/SelectorCompiler.cpp:
(WebCore::SelectorCompiler::addPseudoClassType):

* LayoutTests/fast/css/css-selector-text-expected.txt:
* LayoutTests/fast/css/css-selector-text.html:
* LayoutTests/fast/css/css-set-selector-text-expected.txt:
* LayoutTests/fast/css/css-set-selector-text.html:
* LayoutTests/fast/selectors/invalid-functional-pseudo-class-expected.txt:
* LayoutTests/fast/selectors/invalid-functional-pseudo-class.html:

Canonical link: https://commits.webkit.org/252786@main
